File: webapp/utils/extractToExcel.py
#!/usr/bin/env python3
"""
Extract data fields from test.json and export to Excel
Usage: python3 extractToExcel.py
"""

import json
import pandas as pd
from pathlib import Path

def extract_data_to_excel():
    # Read the JSON file
    print("📖 Reading test.json...")
    with open('test.json', 'r', encoding='utf-8') as f:
        json_data = json.load(f)

    print(f"✅ Loaded {len(json_data)} records")

    # Extract data fields from each object
    extracted_data = []
    for index, item in enumerate(json_data, start=1):
        data_obj = item.get('data', {})
        extracted_data.append({
            'Index': index,
            'ParentId': data_obj.get('ParentId', ''),
            'Parent': data_obj.get('Parent', ''),
            'Child': data_obj.get('Child', ''),
            'RequestUri': item.get('requestUri', ''),
            'Method': item.get('method', '')
        })

    # Create DataFrame
    df = pd.DataFrame(extracted_data)

    # Export to Excel
    output_file = 'extracted_data.xlsx'
    print(f"📝 Writing to {output_file}...")
    df.to_excel(output_file, index=False, sheet_name='Extracted Data')

    print(f"✅ Successfully exported {len(extracted_data)} records to {output_file}")
    print("\n📊 Preview of first 5 rows:")
    print(df.head())

    return output_file

if __name__ == "__main__":
    try:
        output_file = extract_data_to_excel()
        print(f"\n🎉 Done! Open '{output_file}' to view the results.")
    except ImportError as e:
        print("❌ Missing required package.")
        print("Please install pandas and openpyxl:")
        print("  pip3 install pandas openpyxl")
    except FileNotFoundError:
        print("❌ test.json file not found in current directory")
    except Exception as e:
        print(f"❌ Error: {e}")

File: webapp/utils/extractToExcel.js
const fs = require('fs');
const XLSX = require('xlsx');

// Read the JSON file
const jsonData = JSON.parse(fs.readFileSync('./test.json', 'utf8'));

// Extract data fields from each object
const extractedData = jsonData.map((item, index) => {
    return {
        Index: index + 1,
        ParentId: item.data.ParentId || '',
        Parent: item.data.Parent || '',
        Child: item.data.Child || '',
        RequestUri: item.requestUri || '',
        Method: item.method || ''
    };
});

// Create a new workbook
const workbook = XLSX.utils.book_new();

// Convert JSON to worksheet
const worksheet = XLSX.utils.json_to_sheet(extractedData);

// Add the worksheet to the workbook
XLSX.utils.book_append_sheet(workbook, worksheet, 'Extracted Data');

// Write the workbook to an Excel file
XLSX.writeFile(workbook, 'extracted_data.xlsx');

console.log('✅ Data extracted successfully!');
console.log(`📊 Total records: ${extractedData.length}`);
console.log('📁 Output file: extracted_data.xlsx');
